### ISSUE TITLE
Make losses inherit from tf.keras.losses.Loss

### ISSUE THREAD

**Issue by guillaumekln** [guillaumekln/OpenNMT-tf#4]
To check if appropriate as we currently return the numerator and denominator separately.
